apps/omg/lib/omg/state/core.ex

```diff
@@ -455,4 +438,38 @@ defmodule OMG.State.Core do
     is_beginning = tx_index == 0 && Enum.empty?(pending)
     {height, is_beginning}
   end
+
+  @spec db_update_utxos(non_neg_integer(), list(Transaction.Recovered.t())) ::
+          list({:put, :utxo, {Utxo.Position.db_t(), Utxo.t()}} | {:delet, :utxo, Utxo.Position.db_t()})
+  defp db_update_utxos(height, txs) do
+    db_updates_new_utxos =
+      txs
+      |> Enum.with_index()
+      |> Enum.flat_map(fn {tx, tx_idx} -> non_zero_utxos_from(tx, height, tx_idx) end)
+      |> Enum.map(&utxo_to_db_put/1)
+
+    db_updates_spent_utxos =
+      txs
+      |> Enum.flat_map(&Transaction.get_inputs/1)
+      |> Enum.flat_map(fn utxo_pos ->
+        # NOTE: child chain mode don't need 'spend' data for now. Consider to add only in Watcher's modes - OMG-382
+        db_key = Utxo.Position.to_db_key(utxo_pos)
+        [{:delete, :utxo, db_key}, {:put, :spend, {db_key, height}}]
+      end)
+
+    Enum.concat(db_updates_new_utxos, db_updates_spent_utxos)
+  end
+
+  @doc """
+    Filter user utxos from db response.
+    It may take a while for a large response from db
+  """
+  @spec standard_exitable_utxos(list({OMG.DB.utxo_pos_db_t(), OMG.Utxo.t()}), Crypto.address_t()) ::
+          list({Utxo.Position.db_t(), Utxo.t()})
```

Comment:
I don't think this is what get's returned here. Funny that dialyzer wouldn't complain